src/shared/pam-util.c

```diff
@@ -50,18 +50,36 @@ int pam_syslog_pam_error(pam_handle_t *handle, int level, int error, const char
 }
 
 static void cleanup_system_bus(pam_handle_t *handle, void *data, int error_status) {
+        /* The PAM_DATA_SILENT flag is the way that pam_end() communicates to the module stack that this
+         * invocation of pam_end() is not the final one, but in the process that is going to directly exec
+         * the child. This means we are being called after a fork(), and we do not want to try and clean
+         * up the sd-bus object, as the file descriptors (which are CLOEXEC) will be invalid, and we'll hit
+         * an assertion. */
```

Comment:
I don't see how CLOEXEC descriptors could be invalid due to fork.

Reply:
I meant that they get double closed

Reply:
i don't follow. O_CLOEXEC is about execve() not about fork(). Hence I don't get the reference to O_CLOEXEC here. THe fds are going to be fully valid in the child still. We should leave them as-is though, i.e. not invoke read()/write()/shutdown() or so on them since that'd affect the parent too and we don't want that.

I mean, we don't need to be concerned about any fd leak here, because O_CLOEXEC makes sure the fds are eventually closed once the execve() is issued in the child, and maybe that deserves a comment here, but as it is written now I find the comment more confusing than enlightening, because as @ldv-alt pointed out O_CLOEXEC has no affect on fork() but the comment kinda suggests that